From e537fb9bdbd9bbbc612d5521d974d64efbcda0a9 Mon Sep 17 00:00:00 2001
From: Montel Laurent <montel@kde.org>
Date: Wed, 6 Jun 2012 20:40:30 +0200
Subject: [PATCH] Try to fix Bug 283272 - crash when marking folder read

I don't know how status can be empty.
But I fix as it. It will fix crash.
If somebody which has this crash perhaps it can apply this patch and
verify if it fixes bug.

FIXED-IN: 4.8.5
BUG: 283272
(cherry picked from commit 67cfe2290a8c1817a92c6da65da001d267a0f24f)
---
 akonadi/kmime/markascommand.cpp |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/akonadi/kmime/markascommand.cpp b/akonadi/kmime/markascommand.cpp
index 96281d5..e8c5974 100644
--- a/akonadi/kmime/markascommand.cpp
+++ b/akonadi/kmime/markascommand.cpp
@@ -104,7 +104,10 @@ void MarkAsCommand::markMessages()
 
   QSet<QByteArray> flags = mTargetStatus.statusFlags();
   Q_ASSERT( flags.size() == 1 );
-  const Akonadi::Item::Flag flag = *(flags.begin());
+  Akonadi::Item::Flag flag;
+  if(!flags.isEmpty()) 
+    flag = *(flags.begin());
+
   Akonadi::Item::List itemsToModify;
   foreach( const Akonadi::Item &it, mMessages ) {
     Akonadi::Item item( it );
-- 
1.7.10

